Source/WebCore/platform/glib/SystemSettings.cpp

```diff
@@ -0,0 +1,191 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "SystemSettings.h"
+
+#if PLATFORM(GTK) || PLATFORM(WPE)
+
+#include "FontRenderOptions.h"
+#include "RenderTheme.h"
+#include "ScrollbarTheme.h"
+
+namespace WebCore {
+
+SystemSettings& SystemSettings::singleton()
+{
+    static NeverDestroyed<SystemSettings> settings;
+    return settings;
+}
+
+SystemSettings::SystemSettings() = default;
+
+void SystemSettings::settingsChanged(SystemSettings::State&& state, bool applyChanges)
+{
+    bool shouldUpdateStyle = false;
+    bool shouldUpdateScrollbarTheme = false;
+    bool themeSettingsDidChange = false;
+    bool antialiasSettingsDidChange = false;
+    bool hintingSettingsDidChange = false;
+
+    if (state.themeName) {
+        m_state.themeName = WTFMove(state.themeName);
+        shouldUpdateStyle = true;
+        shouldUpdateScrollbarTheme = true;
+    }
+
+    if (state.fontName)
+        m_state.fontName = WTFMove(state.fontName);
+
+    if (state.xftHinting) {
+        m_state.xftHinting = state.xftHinting;
+        hintingSettingsDidChange = true;
+    }
+
+    if (state.xftHintStyle) {
+        m_state.xftHintStyle = WTFMove(state.xftHintStyle);
+        hintingSettingsDidChange = true;
+    }
+
+    if (state.xftAntialias) {
+        m_state.xftAntialias = state.xftAntialias;
+        antialiasSettingsDidChange = true;
+    }
+
+    if (state.xftRGBA) {
+        m_state.xftRGBA = WTFMove(state.xftRGBA);
+        antialiasSettingsDidChange = true;
+    }
+
+    if (antialiasSettingsDidChange || hintingSettingsDidChange)
+        shouldUpdateStyle = true;
+
+    if (state.xftDPI)
+        m_state.xftDPI = state.xftDPI;
+
+    if (state.cursorBlink)
+        m_state.cursorBlink = state.cursorBlink;
+
+    if (state.cursorBlinkTime)
+        m_state.cursorBlinkTime = state.cursorBlinkTime;
+
+    if (state.primaryButtonWarpsSlider)
+        m_state.primaryButtonWarpsSlider = state.primaryButtonWarpsSlider;
+
+    if (state.overlayScrolling) {
+        m_state.overlayScrolling = state.overlayScrolling;
+        shouldUpdateScrollbarTheme = true;
+    }
+
+    if (state.enableAnimations)
+        m_state.enableAnimations = state.enableAnimations;
+
+    if (applyChanges) {
```

Comment:
I thought the plan was to use observers. We can add an observer to the web process to apply these settings, no?